iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tablet/PipeRawTabletInsertionEvent.java

```diff
@@ -107,16 +123,35 @@ public ProgressIndex getProgressIndex() {
 
   @Override
   public EnrichedEvent shallowCopySelfAndBindPipeTaskMetaForProgressReport(
-      String pipeName, PipeTaskMeta pipeTaskMeta, String pattern) {
+      String pipeName, PipeTaskMeta pipeTaskMeta, String pattern, long startTime, long endTime) {
     return new PipeRawTabletInsertionEvent(
-        tablet, isAligned, sourceEvent, needToReport, pipeName, pipeTaskMeta, pattern);
+        tablet,
+        isAligned,
+        sourceEvent,
+        needToReport,
+        pipeName,
+        pipeTaskMeta,
+        pattern,
+        startTime,
+        endTime);
   }
 
   @Override
   public boolean isGeneratedByPipe() {
     throw new UnsupportedOperationException("isGeneratedByPipe() is not supported!");
   }
 
+  @Override
+  public boolean isEventTimeOverlappedWithTimeRange() {
```

Comment:
Elements in tablet.timestamps are sorted. But please consider if tablet.timestamps.length == 0

Reply:
ditto.